code/game/objects/structures/tables_racks.dm

```diff
@@ -335,6 +392,69 @@
 	log_combat(shover, target, "shoved", "onto [src] (table)")
 	return COMSIG_CARBON_SHOVE_HANDLED
 
+/obj/structure/flippedtable
+	name = "flipped table"
+	desc = "A flipped table."
+	icon = 'icons/obj/flipped_tables.dmi'
+	icon_state = "table"
+	anchored = TRUE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	opacity = FALSE
+	///what type of table the flipped table is
+	var/table_type = /obj/structure/table
+
+/obj/structure/flippedtable/Initialize(mapload)
+	. = ..()
+
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_EXIT = PROC_REF(on_exit),
+	)
+
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/flippedtable/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(table_type == /obj/structure/table/glass) //Glass table, jolly ranchers pass
+		if(istype(mover) && (mover.pass_flags & PASSGLASS))
+			return TRUE
+	if(istype(mover, /obj/projectile))
+		var/obj/projectile/projectile = mover
+		//Lets through bullets shot from behind the cover of the table
+		if(projectile.trajectory && angle2dir_cardinal(projectile.trajectory.angle) == dir)
+			return TRUE
+		return FALSE
+	if(border_dir == dir)
+		return FALSE
+	return TRUE
+
+/obj/structure/flippedtable/proc/on_exit(datum/source, atom/movable/leaving, direction)
+	SIGNAL_HANDLER
+
+	if(table_type == /obj/structure/table/glass) //Glass table, jolly ranchers pass
+		if(istype(leaving) && (leaving.pass_flags & PASSGLASS))
+			return
+
+	if(istype(leaving, /obj/projectile))
+		return
```

Comment:
is this the case? is the idea that you can shoot over tables or someodd?

Reply:
Yup, I think the intention was to allow/block shots similar to the sandbags